components/fxa-client/ios/FxAClient/RustFxAccount.swift

```diff
@@ -259,19 +259,28 @@ open class RustFxAccount {
         }
     }
 
-    open func migrateFromSessionToken(sessionToken: String, kSync: String, kXCS: String) throws -> Bool {
-        let json = try nullableRustCall { err in
+    open func migrateFromSessionToken(sessionToken: String, kSync: String, kXCS: String) throws -> String? {
```

Comment:
I know this is what is being done on the Android side, however would it be possible to return a struct instead of a JSON string here. I'd way way prefer to give good type guarantees to our consumers, even if we use JSON to cross the FFI for non perf critical code.

Reply:
@eoger do we have anything that parses JSON in our components?

Reply:
@eoger would we need to pull in `SwiftyJSON` for this?

Reply:
I see `components/logins/ios/Logins/LoginRecord.swift` has an example of doing it, alternatively protobufs is probably preferred here.

Reply:
cool, will take a look